### PR TITLE
Convert Utitlities/zeros* to typescript

### PR DESCRIPTION
**Pull request by harshkhandeparkar**
## Description
- Converted src/utilitities/zeros*.js to typescript
- To use typescript default imports, I either had to do require().default, change the default exports to just exports (*or convert the whole codebase to typescript*)
So I chose option 2 since it is better to [avoid default exports](https://basarat.gitbook.io/typescript/main-1/defaultisbad) anyway.
- Changed all the files that import the zeros* utilities.

NOTE: This branch is based on the `ts` branch so merging #585 is required.

## Motivation and Context

[issue](https://github.com/BrainJS/brain.js/issues/551)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ¯\\\_(ツ)_/¯

## Author's Checklist:

- [x] My code focuses on the main motivation and avoids scope creep.
- [x] My code passes current tests and adds new tests where possible.
- [ ] My code is [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [ ] I have updated the documentation as needed.

## Reviewer's Checklist:
- [ ] I kept my comments to the author positive, specific, and productive.
- [ ] I tested the code and didn't find any new problems.
- [ ] I think the motivation is good for the project.
- [ ] I think the code works to satisfies the motivation.

### CODE REVIEW

File: src/estimator/mean-squared-error.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { IKernelRunShortcut } from 'gpu.js';
+
+/**
+ * 2D Mean Squared Error
+ */
+export function mse2d(
+  this: {
+    constants: { height: number; width: number; length: number };
+  },
+  errors: Array<[number, number]>
```
Comment (harshkhandeparkar) [inline src/estimator/mean-squared-error.ts:11]: Is this the linter? This should have been in the older commits. I believe the linter is running after the commit. Is this a known issue?

**Comment by mubaidr** [comment]
Please pull `master` into `HarshKhandeparkar:js-needs-types`

File: src/estimator/mean-squared-error.ts (hunk 1 above)
Comment (mubaidr) [reply src/estimator/mean-squared-error.ts:11]: Please pull master into HarshKhandeparkar:js-needs-types

**Comment by harshkhandeparkar** [comment]
Okay.

File: src/estimator/mean-squared-error.ts (hunk 1 above)
Comment (harshkhandeparkar) [reply src/estimator/mean-squared-error.ts:11]: Done. This is still there though. Should it happen precommit?

Comment (mubaidr) [reply src/estimator/mean-squared-error.ts:11]: We run `npm run lint` as `pre-commit` hook. So if there is any error/warning in code it will be notified.

**Review by mubaidr — APPROVED** [review]

File: src/estimator/mean-squared-error.ts (hunk 1 above)
Comment (harshkhandeparkar) [reply src/estimator/mean-squared-error.ts:11]: Ohk. I used the VSCode git extension so I didnt get any errors.
If that hook makes any changes, doesn't it commit those changes?

**Pull request merged**